specs/phase1/beacon-chain.md

```diff
@@ -739,311 +433,117 @@ def process_operations(state: BeaconState, body: BeaconBlockBody) -> None:
-    online_indices = get_online_validator_indices(state)
-    shard = compute_shard_from_committee_index(state, committee_index, on_time_attestation_slot)
-
-    # Loop over all shard transition roots
-    shard_transition_roots = set([a.data.shard_transition_root for a in attestations])
-    for shard_transition_root in sorted(shard_transition_roots):
-        transition_attestations = [a for a in attestations if a.data.shard_transition_root == shard_transition_root]
-        transition_participants: Set[ValidatorIndex] = set()
-        for attestation in transition_attestations:
-            participants = get_attesting_indices(state, attestation.data, attestation.aggregation_bits)
-            transition_participants = transition_participants.union(participants)
-
-        enough_online_stake = (
-            get_total_balance(state, online_indices.intersection(transition_participants)) * 3 >=
-            get_total_balance(state, online_indices.intersection(committee)) * 2
-        )
-        # If not enough stake, try next transition root
-        if not enough_online_stake:
-            continue
+    if True in [c.confirmed for c in all_candidates]:
+        return
 
-        # Attestation <-> shard transition consistency
-        assert shard_transition_root == hash_tree_root(shard_transition)
-
-        # Check `shard_head_root` of the winning root
-        last_offset_index = len(shard_transition.shard_states) - 1
-        shard_head_root = shard_transition.shard_states[last_offset_index].latest_block_root
-        for attestation in transition_attestations:
-            assert attestation.data.shard_head_root == shard_head_root
-
-        # Apply transition
-        apply_shard_transition(state, shard, shard_transition)
-        # Apply proposer reward and cost
-        beacon_proposer_index = get_beacon_proposer_index(state)
-        estimated_attester_reward = sum([get_base_reward(state, attester) for attester in transition_participants])
-        proposer_reward = Gwei(estimated_attester_reward // PROPOSER_REWARD_QUOTIENT)
-        increase_balance(state, beacon_proposer_index, proposer_reward)
-        states_slots_lengths = zip(
-            shard_transition.shard_states,
-            get_offset_slots(state, shard),
-            shard_transition.shard_block_lengths
-        )
-        for shard_state, slot, length in states_slots_lengths:
-            proposer_index = get_shard_proposer_index(state, slot, shard)
-            decrease_balance(state, proposer_index, shard_state.gasprice * length)
-
-        # Return winning transition root
-        return shard_transition_root
-
-    # No winning transition root, ensure empty and return empty root
-    assert shard_transition == ShardTransition()
-    return Root()
+    # Requirement 2: >= 2/3 of balance attesting
+    participants = get_attesting_indices(state, attestation.data, pending_commitment.votes)
```

Comment:
> `pending_commitment.votes` 

Shouldn't this be `pending_header.votes` ?

Reply:
yes! thank you
fixed